crypto/dsa/dsa_check.c

```diff
@@ -19,8 +19,34 @@
 #include "dsa_local.h"
 #include "crypto/dsa.h"
 
+static int dsa_precheck_params(const DSA *dsa, int *ret)
+{
+    if (dsa->params.p == NULL || dsa->params.q == NULL) {
```

Comment:
I dont think this check for q being NULL is correct.. shouldnt it just be part of the check on line 36..

Reply:
See ossl_dsa_check_priv_key(),
and the pairwise doesnt use q either.

Reply:
But the q param is mandatory in DSA parameters (not DH parameters!). I do not see anything wrong with the precheck enforcing mandatory params.

Reply:
And IMO whether it is placed here or at line 36 is a matter of taste. I prefer it like this.

Reply:
I thought the point of the check was for bounds, not to enforce it being non NULL..

The difference is if (q != NULL && q < )

Reply:
You should also get static analysis hits with your change, because of the check in  ossl_dsa_check_priv_key

Reply:
I can remove that redundant check.